### PR TITLE
Story/support start form

### PR DESCRIPTION
**Pull request by Tom-Ritense**

### CODE REVIEW

File: form-view-model/src/main/kotlin/com/ritense/formviewmodel/autoconfigure/FormViewModelAutoConfiguration.kt [hunk 1]
```diff
@@ -78,7 +81,8 @@ class FormViewModelAutoConfiguration {
         camundaTaskService: CamundaTaskService,
         authorizationService: AuthorizationService,
         formViewModelService: FormViewModelService,
-        formViewModelSubmissionService: FormViewModelSubmissionService
+        formViewModelSubmissionService: FormViewModelSubmissionService,
+        formDefinitionService: FormIoFormDefinitionService
```
Comment (laurens-ritense) [inline form-view-model/src/main/kotlin/com/ritense/formviewmodel/autoconfigure/FormViewModelAutoConfiguration.kt]: Is this still necessary?

File: form-view-model/src/main/kotlin/com/ritense/formviewmodel/web/rest/FormViewModelResource.kt [hunk 2]
```diff
@@ -66,40 +82,69 @@ class FormViewModelResource(
...
+            viewModelLoaderFactory.getViewModelLoader(formName) ?: return ResponseEntity.notFound().build()
+        val viewModelType = viewModelLoader.getViewModelType()
+        val updatedViewModel = formViewModelService.parseViewModel(submission, viewModelType).update()
+        return ResponseEntity.ok(updatedViewModel)
+    }
+
+    @PostMapping("/user-task")
+    fun updateUserTaskFormViewModel(
+        @RequestParam formName: String,
+        @RequestParam taskInstanceId: String? = null,
```
Comment (laurens-ritense) [inline form-view-model/src/main/kotlin/com/ritense/formviewmodel/web/rest/FormViewModelResource.kt]: taskInstanceId should not be optional in the request param

File: form-view-model/src/main/kotlin/com/ritense/formviewmodel/web/rest/FormViewModelResource.kt [hunk 3]
```diff
@@ -66,40 +82,69 @@ class FormViewModelResource(
...
 
-    @PostMapping("/submit")
-    @Transactional
-    fun submit(
-        @RequestParam(required = true) formName: String,
-        @RequestParam(required = true) taskInstanceId: String,
+    @PostMapping("/submit/user-task")
+    fun submitTask(
+        @RequestParam formName: String,
+        @RequestParam taskInstanceId: String? = null,
```
Comment (laurens-ritense) [inline form-view-model/src/main/kotlin/com/ritense/formviewmodel/web/rest/FormViewModelResource.kt]: again

File: form-view-model/src/main/kotlin/com/ritense/formviewmodel/event/FormViewModelSubmissionHandler.kt [hunk 4]
```diff
@@ -26,7 +26,7 @@ interface FormViewModelSubmissionHandler<T : Submission> {
```
Comment (laurens-ritense) [inline form-view-model/src/main/kotlin/com/ritense/formviewmodel/event/FormViewModelSubmissionHandler.kt:26]: rename this package to submission

File: form-view-model/src/main/kotlin/com/ritense/formviewmodel/service/FormViewModelService.kt [hunk 5]
```diff
@@ -22,61 +22,58 @@ import com.ritense.authorization.AuthorizationService
...
-    private val authorizationService: AuthorizationService,
-    private val formViewModelSubmissionService: FormViewModelSubmissionService
+    private val authorizationService: AuthorizationService
 ) {
-    fun getFormViewModel(
+
+    fun getStartFormViewModel(
+        formName: String,
+    ): ViewModel? {
+        return viewModelLoaderFactory.getViewModelLoader(formName)?.load()
```
Comment (laurens-ritense) [inline form-view-model/src/main/kotlin/com/ritense/formviewmodel/service/FormViewModelService.kt:55]: No permissions check?

Comment (Tom-Ritense) [reply form-view-model/src/main/kotlin/com/ritense/formviewmodel/service/FormViewModelService.kt:55]: I didn't know which one....

Comment (laurens-ritense) [reply form-view-model/src/main/kotlin/com/ritense/formviewmodel/service/FormViewModelService.kt:55]: Ask

**Review by laurens-ritense — APPROVED** [review]

**Pull request merged**